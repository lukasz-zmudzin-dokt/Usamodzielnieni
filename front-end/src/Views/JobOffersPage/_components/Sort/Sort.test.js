import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { Sort } from "../";

describe("Sort-JobOffersPage", () => {
  let props = {
    sort: "salary_min",
    setSort: (val) => val,
  };

  it("should render withoud crashing", () => {
    const { container } = render(<Sort {...props} />);
    expect(container).toMatchSnapshot();
  });

  it("should change check on click", () => {
    const { getByTestId } = render(<Sort {...props} />);

    fireEvent.click(getByTestId("checkUp-salary_min--sort"));
    expect(getByTestId("checkUp-salary_min--sort").checked).toBe(true);

    fireEvent.click(getByTestId("checkUp-salary_min--sort"));
    expect(getByTestId("checkUp-salary_min--sort").checked).toBe(false);
  });
});
