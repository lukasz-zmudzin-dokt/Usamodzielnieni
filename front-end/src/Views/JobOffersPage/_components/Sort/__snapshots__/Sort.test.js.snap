// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sort-JobOffersPage should render withoud crashing 1`] = `
<div>
  <div
    class="accordion"
  >
    <div
      class="card"
    >
      <div
        class="card-header"
      >
        <button
          class="btn btn-link"
          type="button"
        >
          <div
            class="m-0 card-title h5"
          >
            Sortowanie
          </div>
        </button>
      </div>
      <div
        class="collapse"
      >
        <div
          class="card-body"
        >
          <div
            class="form-row"
          >
            <div
              class="form-group col-md-3 col-12"
            >
              <label
                class="form-label"
                for="offer_name--sort"
              >
                Nazwa oferty
              </label>
              <div>
                <div
                  class="form-check form-check-inline"
                >
                  <input
                    class="form-check-input"
                    data-testid="checkUp-offer_name--sort"
                    id="inline-checkbox-Nazwa oferty1"
                    type="checkbox"
                  />
                  <label
                    class="form-check-label"
                    for="inline-checkbox-Nazwa oferty1"
                    title=""
                  >
                    Rosnąco
                  </label>
                </div>
                <div
                  class="form-check form-check-inline"
                >
                  <input
                    class="form-check-input"
                    data-testid="checkDown-offer_name--sort"
                    id="inline-checkbox-Nazwa oferty2"
                    type="checkbox"
                  />
                  <label
                    class="form-check-label"
                    for="inline-checkbox-Nazwa oferty2"
                    title=""
                  >
                    Malejąco
                  </label>
                </div>
              </div>
            </div>
            <div
              class="form-group col-md-3 col-12"
            >
              <label
                class="form-label"
                for="category--sort"
              >
                Branża
              </label>
              <div>
                <div
                  class="form-check form-check-inline"
                >
                  <input
                    class="form-check-input"
                    data-testid="checkUp-category--sort"
                    id="inline-checkbox-Branża1"
                    type="checkbox"
                  />
                  <label
                    class="form-check-label"
                    for="inline-checkbox-Branża1"
                    title=""
                  >
                    Rosnąco
                  </label>
                </div>
                <div
                  class="form-check form-check-inline"
                >
                  <input
                    class="form-check-input"
                    data-testid="checkDown-category--sort"
                    id="inline-checkbox-Branża2"
                    type="checkbox"
                  />
                  <label
                    class="form-check-label"
                    for="inline-checkbox-Branża2"
                    title=""
                  >
                    Malejąco
                  </label>
                </div>
              </div>
            </div>
            <div
              class="form-group col-md-3 col-12"
            >
              <label
                class="form-label"
                for="voivodeship--sort"
              >
                Województwo
              </label>
              <div>
                <div
                  class="form-check form-check-inline"
                >
                  <input
                    class="form-check-input"
                    data-testid="checkUp-voivodeship--sort"
                    id="inline-checkbox-Województwo1"
                    type="checkbox"
                  />
                  <label
                    class="form-check-label"
                    for="inline-checkbox-Województwo1"
                    title=""
                  >
                    Rosnąco
                  </label>
                </div>
                <div
                  class="form-check form-check-inline"
                >
                  <input
                    class="form-check-input"
                    data-testid="checkDown-voivodeship--sort"
                    id="inline-checkbox-Województwo2"
                    type="checkbox"
                  />
                  <label
                    class="form-check-label"
                    for="inline-checkbox-Województwo2"
                    title=""
                  >
                    Malejąco
                  </label>
                </div>
              </div>
            </div>
            <div
              class="form-group col-md-3 col-12"
            >
              <label
                class="form-label"
                for="company_name--sort"
              >
                Nazwa firmy
              </label>
              <div>
                <div
                  class="form-check form-check-inline"
                >
                  <input
                    class="form-check-input"
                    data-testid="checkUp-company_name--sort"
                    id="inline-checkbox-Nazwa firmy1"
                    type="checkbox"
                  />
                  <label
                    class="form-check-label"
                    for="inline-checkbox-Nazwa firmy1"
                    title=""
                  >
                    Rosnąco
                  </label>
                </div>
                <div
                  class="form-check form-check-inline"
                >
                  <input
                    class="form-check-input"
                    data-testid="checkDown-company_name--sort"
                    id="inline-checkbox-Nazwa firmy2"
                    type="checkbox"
                  />
                  <label
                    class="form-check-label"
                    for="inline-checkbox-Nazwa firmy2"
                    title=""
                  >
                    Malejąco
                  </label>
                </div>
              </div>
            </div>
            <div
              class="form-group col-md-3 col-12"
            >
              <label
                class="form-label"
                for="expiration_date--sort"
              >
                Okres ważności
              </label>
              <div>
                <div
                  class="form-check form-check-inline"
                >
                  <input
                    class="form-check-input"
                    data-testid="checkUp-expiration_date--sort"
                    id="inline-checkbox-Okres ważności1"
                    type="checkbox"
                  />
                  <label
                    class="form-check-label"
                    for="inline-checkbox-Okres ważności1"
                    title=""
                  >
                    Rosnąco
                  </label>
                </div>
                <div
                  class="form-check form-check-inline"
                >
                  <input
                    class="form-check-input"
                    data-testid="checkDown-expiration_date--sort"
                    id="inline-checkbox-Okres ważności2"
                    type="checkbox"
                  />
                  <label
                    class="form-check-label"
                    for="inline-checkbox-Okres ważności2"
                    title=""
                  >
                    Malejąco
                  </label>
                </div>
              </div>
            </div>
            <div
              class="form-group col-md-3 col-12"
            >
              <label
                class="form-label"
                for="salary_min--sort"
              >
                Minimalne wynagrodzenie
              </label>
              <div>
                <div
                  class="form-check form-check-inline"
                >
                  <input
                    class="form-check-input"
                    data-testid="checkUp-salary_min--sort"
                    id="inline-checkbox-Minimalne wynagrodzenie1"
                    type="checkbox"
                  />
                  <label
                    class="form-check-label"
                    for="inline-checkbox-Minimalne wynagrodzenie1"
                    title=""
                  >
                    Rosnąco
                  </label>
                </div>
                <div
                  class="form-check form-check-inline"
                >
                  <input
                    class="form-check-input"
                    data-testid="checkDown-salary_min--sort"
                    id="inline-checkbox-Minimalne wynagrodzenie2"
                    type="checkbox"
                  />
                  <label
                    class="form-check-label"
                    for="inline-checkbox-Minimalne wynagrodzenie2"
                    title=""
                  >
                    Malejąco
                  </label>
                </div>
              </div>
            </div>
            <div
              class="form-group col-md-3 col-12"
            >
              <label
                class="form-label"
                for="salary_max--sort"
              >
                Maksymalne wynagrodzenie
              </label>
              <div>
                <div
                  class="form-check form-check-inline"
                >
                  <input
                    class="form-check-input"
                    data-testid="checkUp-salary_max--sort"
                    id="inline-checkbox-Maksymalne wynagrodzenie1"
                    type="checkbox"
                  />
                  <label
                    class="form-check-label"
                    for="inline-checkbox-Maksymalne wynagrodzenie1"
                    title=""
                  >
                    Rosnąco
                  </label>
                </div>
                <div
                  class="form-check form-check-inline"
                >
                  <input
                    class="form-check-input"
                    data-testid="checkDown-salary_max--sort"
                    id="inline-checkbox-Maksymalne wynagrodzenie2"
                    type="checkbox"
                  />
                  <label
                    class="form-check-label"
                    for="inline-checkbox-Maksymalne wynagrodzenie2"
                    title=""
                  >
                    Malejąco
                  </label>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
